cloud/services/compute/loadbalancers/reconcile.go

```diff
@@ -234,7 +234,7 @@ func (s *Service) createInternalLoadBalancer(ctx context.Context, name string, l
 	if err != nil {
 		return err
 	}
-	s.scope.Network().APIInternalAddress = ptr.To[string](addr.Address)
+	s.scope.Network().APIInternalAddress = ptr.To[string](addr.SelfLink)
```

Comment:
Is SelfLink more appropriate than Address in this case?
I not an expert on this, what is the difference?
Thanks!

Reply:
This provides a url to the resource and is consistent with how the other fields are handled particularly apiServerIpAddress. With the url, a consumer can retrieve the resource and get and fields including IP Address. For example, with this fix the following status fields are set when using dual Load Balancers:
```
network:
    apiInternalBackendService: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/regions/us-east4/backendServices/capg-e2e-y0908n-api-internal
    apiInternalForwardingRule: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/regions/us-east4/forwardingRules/capg-e2e-y0908n-api-internal
    apiInternalHealthCheck: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/regions/us-east4/healthChecks/capg-e2e-y0908n-api-internal
    apiInternalIpAddress: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/regions/us-east4/addresses/capg-e2e-y0908n-api-internal
    apiServerBackendService: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/global/backendServices/capg-e2e-y0908n-apiserver
    apiServerForwardingRule: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/global/forwardingRules/capg-e2e-y0908n-apiserver
    apiServerHealthCheck: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/global/healthChecks/capg-e2e-y0908n-apiserver
    apiServerInstanceGroups:
      us-east4-a: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/zones/us-east4-a/instanceGroups/capg-e2e-y0908n-apiserver-us-east4-a
      us-east4-b: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/zones/us-east4-b/instanceGroups/capg-e2e-y0908n-apiserver-us-east4-b
      us-east4-c: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/zones/us-east4-c/instanceGroups/capg-e2e-y0908n-apiserver-us-east4-c
    apiServerIpAddress: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/global/addresses/capg-e2e-y0908n-apiserver
    apiServerTargetProxy: https://www.googleapis.com/compute/v1/projects/k8s-infra-e2e-boskos-122/global/targetTcpProxies/capg-e2e-y0908n-apiserver
```

Without this fix, this field would be (for example):
`apiInternalIpAddress: 10.0.0.2`

Reply:
Ah! Got it thanks!